version: "3"

services:
  proxy:
    image: traefik:v2.10
    command:
      - "--api.insecure=true"
      - "--providers.docker"
      - "--providers.file.directory=/etc/traefik/dynamic"
      - "--log.level=DEBUG"
      - "--entryPoints.http.address=:80"
      - "--entryPoints.http.http.redirections.entryPoint.to=https"
      - "--entryPoints.http.http.redirections.entryPoint.scheme=https"
      - "--entryPoints.https.address=:443"
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./certs-traefik.yml:/etc/traefik/dynamic/certs-traefik.yml
      - ./certs:/etc/certs

  web:
    build:
      context: .
      dockerfile: Dockerfile
    tty: true
    volumes:
      - ..:/workspace:cached
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.web.tls=true"
      - "traefik.http.routers.web.rule=Host(`dev.zoneblitz.app`)"

  database:
    image: postgres:16
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: dev
      POSTGRES_PASSWORD: password
      POSTGRES_DB: zone_blitz
    labels:
      - "traefik.enable=false"

  database-admin:
    image: dpage/pgadmin4:latest
    ports:
      - "3000:80"
    environment:
      PGADMIN_DEFAULT_EMAIL: "dev@dev.zoneblitz.app"
      PGADMIN_DEFAULT_PASSWORD: "password"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    labels:
      - "traefik.enable=false"
